core/src/main/java/org/infinispan/interceptors/impl/DistCacheWriterInterceptor.java

```diff
@@ -88,7 +91,9 @@ public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) t
          PutMapCommand putMapCommand = (PutMapCommand) rCommand;
          LocalizedCacheTopology cacheTopology = dm.getCacheTopology();
          Map<Object, Object> map = putMapCommand.getMap();
-         int count = 0;
+
```

Comment:
The lambda implicitly captures `this` (for example to be able to call `isProperWriter`) and that results in an allocation. Regrettably JVM does not optimize this out yet.